src/EditorFeatures/Core/Classification/CopyPasteAndPrintingClassificationBufferTaggerProvider.Tagger.cs

```diff
@@ -154,9 +154,9 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetAllTags(NormalizedSnapshotSp
                     }
                 }
 
-                return cachedTags == null
-                    ? Array.Empty<ITagSpan<IClassificationTag>>()
-                    : cachedTags.GetIntersectingTagSpans(spans);
+                using var pooledTags = Classifier.GetPooledList<ITagSpan<IClassificationTag>>(out var tags);
```

Comment:
>using [](http://example.com/codeflow?start=16&length=5)

this using shouldn't be here, right?

Reply:
(It kind of make me wonder whether this is going to be a common problem with using GetFinalList and whether it's just better to accept the 0 item case as doing an allocation)

Reply:
great catch.

Reply:
changed this to a cleaner pattern that doesn't depend on using/GetFinalList.